src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs

```diff
@@ -254,23 +257,36 @@ public string GetLastBackupTaken()
 
         }
 
+        public override void Execute(TaskExecutionMode mode)
+        {
+            UpdateRestoreTaskObject();
+
+            base.Execute(mode);
+        }
+
         /// <summary>
         /// Executes the restore operations
         /// </summary>
-        public void Execute()
+        public override void Execute()
         {
-            RestorePlan restorePlan = GetRestorePlanForExecutionAndScript();
-            
-            if (restorePlan != null && restorePlan.RestoreOperations.Count > 0)
+            if (IsValid && RestorePlan.RestoreOperations != null && RestorePlan.RestoreOperations.Any())
             {
-                restorePlan.PercentComplete += (object sender, PercentCompleteEventArgs e) =>
+                // Restore Plan should be already created and updated at this point
+
+                RestorePlan restorePlan = GetRestorePlanForExecutionAndScript();
+
+                if (restorePlan != null && restorePlan.RestoreOperations.Count > 0)
                 {
-                    if (SqlTask != null)
+                    restorePlan.PercentComplete += (object sender, PercentCompleteEventArgs e) =>
```

Comment:
not yet, the task manager needs to show the messages

---
In reply to: [134043131](https://github.com/Microsoft/sqltoolsservice/pull/438#discussion_r134043131) [](ancestors = 134043131)